types/puppeteer/index.d.ts

```diff
@@ -448,22 +451,69 @@ export interface Box {
   y: number;
 }
 
+/**
+ * The Worker class represents a WebWorker.
+ * The events workercreated and workerdestroyed are emitted on the page object to signal the worker lifecycle.
+ */
+export interface Worker {
+  /**
+   * If the function passed to the `worker.evaluate` returns a Promise,
+   * then `worker.evaluate` would wait for the promise to resolve and return its value.
+   *
+   * If the function passed to the `worker.evaluate` returns a non-Serializable value,
+   * then `worker.evaluate` resolves to `undefined`.
+   */
+  evaluate<T>(
```

Comment:
It _might_ be worth considering a couple of overloads for this function (and other evaluation funcs):
```typescript
evaluate<R>(pageFunction: () => R | Promise<R>): Promise<R>;
evaluate<R, X1>(pageFunction: (x1: X1) => T | Promise<T>, x1: X1): Promise<T>;
...
evaluate<R>(pageFunction: (...args: any[]) => R | Promise<R>, ...args: any[]): Promise<R>;
```

Reply:
Actually this may be problematic when you pass an `ElementHandle` to `evaluate`'s args, because in the callback it will be an actual `HTMLElement` of the DOM.

Reply:
Hmm, that's an interesting point. With conditional mapped types in ts2.8, would be easy to solve :) But that might be more trouble than it is worth.

Reply:
@andnp I'm all for bumping TS version to 2.8 if it improves type-safety 👍